File: ackermann_function.swift
// http://rosettacode.org/wiki/Ackermann_function

func ackermann(m:Int,n:Int) -> Int {
  if m == 0 {
    return n + 1
  } else if m > 0 && n == 0 {
    return ackermann(m-1,1)
  } else if m > 0 && n > 0 {
    return ackermann(m-1,(ackermann(m,n-1)))
  } else {
    return -10
  }
}

for var m = 0; m < 4; m++ {
  for var n = 0; n < 4; n++ {
    let result = ackermann(m,n)
    println("\(m), \(n): \(result)")
  }
}

File: ackermann_function.py
#!/usr/bin/env python
# http://rosettacode.org/wiki/Ackermann_function

def ackermann(m,n):
  if m == 0:
    return n + 1
  elif m > 0 and n == 0:
    return ackermann(m-1,1)
  elif m > 0 and n > 0:
    return ackermann(m-1,(ackermann(m,n-1)))

for m in xrange(0,4):
  for n in xrange(0,4):
    print str(m) + ", " + str(n) + ": " + str(ackermann(m,n))
